src/sfdp.c

```diff
@@ -143,6 +144,57 @@ static int read_jedec_flash_parameter(FLASHER_SPI_FLASH_T *ptFlash, unsigned lon
 	{
 		hexdump(uSfdpData.auc, sizeof(uSfdpData.auc));
 
+		/** Read erase instruction entries from JEDEC Basic Flash Parametes
+		 *  Read starting from DWORD 8, there are 4 erase entries each with:
+		 *  8-Bit for the size in 2^N format
+		 *  8-Bit for the OpCode
+		 */
+		const unsigned char DWORD_8_START = 7*4; /* Nr of DWords starts at 1*/
+		unsigned int nrEraseOps = 0; /* Required for sorting */
+
+		/* Get erase operations from SFDP data */
+		for(int opNr = 0; opNr < FLASHER_SPI_NR_ERASE_INSTRUCTIONS; opNr++){
+			unsigned char shift = uSfdpData.auc[DWORD_8_START + 2*opNr];
+			unsigned char opCode = uSfdpData.auc[DWORD_8_START + 2*opNr + 1];
+			/* No instruction present or shift further than variable size */
+			if(shift == 0 || shift >= sizeof(unsigned long long)*8){
+				ptFlash->tSpiErase[opNr].Size = 0;
+				ptFlash->tSpiErase[opNr].OpCode = 0;
+			}else{
+				ptFlash->tSpiErase[opNr].Size = 1U << shift;
+				ptFlash->tSpiErase[opNr].OpCode = opCode;
+				nrEraseOps++;
+			}
+		}
+
+		/* Sort the erase commands so the smallest is in element 0 */
+		for(unsigned int sortPos = 0; sortPos < nrEraseOps; sortPos++){
+			unsigned long tmpEraseSize = ptFlash->tSpiErase[sortPos].Size;
+			unsigned char tmpEraseOpCode = ptFlash->tSpiErase[sortPos].OpCode;
+			unsigned int tmpMinPos = sortPos;
+			for(unsigned int eraseCmd = sortPos + 1; eraseCmd < nrEraseOps; eraseCmd++)
+			{
+				if(ptFlash->tSpiErase[eraseCmd].Size <= tmpEraseSize)
+				{
+					tmpEraseSize = ptFlash->tSpiErase[eraseCmd].Size;
+					tmpEraseOpCode = ptFlash->tSpiErase[eraseCmd].OpCode;
+					tmpMinPos = eraseCmd;
+				}
+			}
+			tmpEraseSize = ptFlash->tSpiErase[sortPos].Size;
+			tmpEraseOpCode = ptFlash->tSpiErase[sortPos].OpCode;
+			ptFlash->tSpiErase[sortPos] = ptFlash->tSpiErase[tmpMinPos];
+			ptFlash->tSpiErase[tmpMinPos].Size = tmpEraseSize;
+			ptFlash->tSpiErase[tmpMinPos].OpCode = tmpEraseOpCode;
+		}
```

Comment:
This is more convoluted than it needs to be. It took me a while to understand.
How about 
FLASHER_SPI_ERASE_T tTmpErase;
unsigned int min_pos;
for (unsigned int pos1 = 0; pos1 < nrEraseOps; pos1++){
	min_pos = pos1;
	for (unsigned int pos2 = pos1+1; pos2 < nrEraseOps; pos2++){
		if (ptFlash->tSpiErase[pos2].Size < ptFlash->tSpiErase[min_pos].Size) {
			min_pos = pos2;
		}
	}
	
	tTmpErase = ptFlash->tSpiErase[pos1];
	ptFlash->tSpiErase[pos1] = ptFlash->tSpiErase[min_pos];
	ptFlash->tSpiErase[min_pos] = tTmpErase;
}

Or even plain bubblesort.